### DISCUSSION TITLE
User permission to audio device

### DISCUSSION THREAD

**Discussion by Venikeee1** [Jelmerro/Vieb#423] [Q&A] [answered]
I am working on voice recorder app. And I can't find how can I confirm permission to get access to my mike

**Comment by Jelmerro** [comment, marked as answer]
Please have a look at `:h permissionmediadevices` and `:h permissionmicrophone`. You need to give permission to both, since the first will allow the site to list the devices, and the second one to make use of the mic. If you only want to give permissions to specific sites, check out `:h permissionsallowed`. Some sites want to have the device labels to work as well, in which case you can replace `allow` with `allowfull` for the media devices permission, but in short, this should be enough:
```vim
:set permissionmediadevices=allow permissionmicrophone=allow
```
Please know that you can always see the permission requests on the notification page, either visit `vieb://notifications` or run `:notifications`. This can help a lot with seeing what is happening in terms of permissions. In case you want to be notified when pages request permissions, please see run `:set notificationforpermissions`, though keep in mind that happens a lot on some pages. Let me know if you still have questions.
